src/rosdistro/freeze_source.py

```diff
@@ -91,11 +91,11 @@ def _worker(work_queue):
             ls_remote_lines = subprocess.check_output(cmd).splitlines()
             for line in ls_remote_lines:
                 hash, ref = line.split('\t', 1)
-                if ref.endswith(freeze_version):
-                    if freeze_to_tag and ref.startswith('refs/tags/'):
-                        source_repo.version = ref.split('refs/tags/')[1]
-                    else:
-                        source_repo.version = hash
+                if ref == 'refs/tags/%s' % freeze_version and freeze_to_tag:
```

Comment:
I think swapping the order makes the condition a bit easier to read, first the important flag then the "messy" detail:

```
if freeze_to_tag and ref == 'refs/tags/%s' % freeze_version:
```